src/primo/browzine-primo-adapter.js

```diff
@@ -276,6 +333,10 @@ browzine.primo = (function() {
     return $scope && $scope.$ctrl && $scope.$ctrl.parentCtrl ? $scope.$ctrl.parentCtrl : undefined;
   };
 
+  function shouldEnhance(scope) {
+    return !!(!isFiltered(scope) && ((isJournal(scope) && getIssn(scope)) || (isArticle(scope) && getDoi(scope))));
```

Comment:
Could we break out the logic on this line into separate steps? I had some difficulty parsing it with the level of parentheses nesting. A couple alternatives that helped me understand the logic:

Split the line:

```js
return !!(
  !isFiltered(scope)
  && (
    (
      isJournal(scope)
      && getIssn(scope)
    )
    || (
      isArticle(scope)
      && getDoi(scope)
    )
  )
);
```

Or use short circuit logic (this should be functionally equivalent):

```js
if (isFiltered(scope)) return false;

if (isJournal(scope) && getIssn(scope)) return true;

if (isArticle(scope) && getDoi(scope)) return true;

return false;
```

Reply:
Yep! It's time. Will look into it. Thanks!

Reply:
Agreed, a little reformatting here could speed up comprehending this logic.